### PR TITLE
Add dapp transaction - Closes #887

### LINKED ISSUE #887: Add DappTransaction class
### Description

Extend `BaseTransaction` implement all the specific features of dapp registration transactions.

### PR DESCRIPTION
**Pull request by shuse2**
### Description
Adding Dapp transaction with feature of handling.

### Review checklist

* The PR resolves #887 
* Merge after #976 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-transactions/src/transactions/5_dapp_transaction.ts [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+				},
+				name: {
+					type: 'string',
+				},
+			},
+		},
+	},
+};
+
+export const dappAssetFormatSchema = {
```
Comment (SargeKhan) [inline packages/lisk-transactions/src/transactions/5_dapp_transaction.ts:108]: Do the rest of the properties have no min/max length constraint?

**Review by SargeKhan — COMMENTED** [review]
Is there no apply and undo functions for DAPP transactions?

**Comment by shuse2** [comment]
@SargeKhan I think Dapp transaction doesn't affect the account except for sender fee, so no need for extending apply and undo

**Review by SargeKhan — APPROVED** [review]

File: packages/lisk-transactions/src/transactions/5_dapp_transaction.ts [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+	RequiredState,
+	TransactionResponse,
+} from './base';
+
+export interface DappAsset {
+	readonly dapp: {
+		readonly category: number;
+		readonly description?: string;
+		readonly icon?: string;
+		readonly link?: string;
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/5_dapp_transaction.ts:41]: `link` is required?

File: packages/lisk-transactions/src/transactions/5_dapp_transaction.ts [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+	},
+};
+
+export const dappAssetFormatSchema = {
+	type: 'object',
+	required: ['dapp'],
+	properties: {
+		dapp: {
+			type: 'object',
+			required: ['name', 'type', 'category'],
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/5_dapp_transaction.ts:114]: same as above

File: packages/lisk-transactions/src/transactions/5_dapp_transaction.ts [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+
+		if (!passphrase) {
+			return transaction;
+		}
+
+		const transactionWithSenderInfo = {
+			...transaction,
+			senderId: transaction.senderId as string,
+			senderPublicKey: transaction.senderPublicKey as string,
+			recipientId: transaction.senderId as string,
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/5_dapp_transaction.ts]: why set `recipientId` to `senderId` here?

File: packages/lisk-transactions/src/transactions/5_dapp_transaction.ts [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+			[ENTITY_TRANSACTION]: uniqueFields,
+		};
+	}
+
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const sameTypeTransactions = transactions
+			.filter(tx => tx.type === this.type)
+			.map(tx => new DappTransaction(tx));
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/5_dapp_transaction.ts]: necessary to initialize here? at this stage will `transactions` already be type checked?

File: packages/lisk-transactions/src/transactions/5_dapp_transaction.ts (hunk 2 above)
Comment (shuse2) [reply packages/lisk-transactions/src/transactions/5_dapp_transaction.ts:41]: https://github.com/LiskHQ/lisk/blob/development/logic/dapp.js#L423

seeing this, it seems like link is optional

**Comment by shuse2** [comment]
@mitsujutsu Addressed

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**